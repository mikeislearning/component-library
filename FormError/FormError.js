import React from 'react';
import { Image, Text, View } from 'react-native';
import { PropTypes } from 'prop-types';
import { images } from 'app/theme';

import styles from './FormError.styles';

export default class FormError extends React.Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
  };

  render() {
    const { text } = this.props;
    if (!text) {
      return null;
    }

    return (
      <View style={styles.container}>
        <Image source={images.error} style={styles.icon} />
        <Text style={styles.text}>{text}</Text>
      </View>
    );
  }
}
