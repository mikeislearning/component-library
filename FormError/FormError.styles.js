import { StyleSheet } from 'react-native';

import { colors, fonts } from 'app/theme';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginTop: 1,
  },
  icon: {
    height: 16,
    marginTop: 3,
    width: 16,
  },
  text: {
    color: colors.red.light,
    fontFamily: fonts.family.messinaSans.regular,
    fontSize: 16,
    marginLeft: 3,
    marginTop: 2,
  },
});

export default styles;
