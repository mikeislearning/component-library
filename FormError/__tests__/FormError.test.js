import React from 'react';
import { render } from 'react-native-testing-library';

import FormError from '../FormError';

describe('<FormError />', () => {
  it('renders correctly', () => {
    const error = render(<FormError text={'Error message'} />);
    expect(error).toMatchSnapshot();
  });

  it('renders nothing if text is not passed', () => {
    const error = render(<FormError />);
    expect(error).toMatchSnapshot();
  });
});
