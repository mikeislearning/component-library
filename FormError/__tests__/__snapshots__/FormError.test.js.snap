// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormError /> renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/images/error.png",
      }
    }
    style={
      Object {
        "height": 16,
        "marginTop": 3,
        "width": 16,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#FF001F",
        "fontFamily": "MessinaSans-Regular",
        "fontSize": 16,
        "marginLeft": 3,
        "marginTop": 2,
      }
    }
  >
    Error message
  </Text>
</View>
`;

exports[`<FormError /> renders nothing if text is not passed 1`] = `null`;
